app/src/main/java/org/fossasia/pslab/fragment/LALogicLinesFragment.java

```diff
@@ -45,9 +49,13 @@
     private ScienceLab scienceLab;
     private LineChart logicLinesChart;
     private ArrayList<String> channelNames = new ArrayList<>();
+    private ArrayList<String> edgesNames = new ArrayList<>();
     private TextView tvTimeUnit;
     private ImageView ledImageView;
     private Runnable logicAnalysis;
+    private TextView channel1_text, channel2_text, channel3_text, channel4_text;
+    private LinearLayout Channel1, Channel2, Channel3, Channel4;
+    private String[] edges = {"EVERY EDGE", "EVERY FALLING EDGE", "EVERY RISING EDGE", "EVERY FOURTH RISING EDGE", "DISABLED"};
```

Comment:
Attributes for the Spinner can be included in the resource file and populate it using that